.github/workflows/opensearch-gha-jobs-lambda.yml

```diff
@@ -0,0 +1,44 @@
+name: Test and deploy opensearch-gha-jobs lambda
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/opensearch-gha-jobs-lambda.yml
+      - aws/lambda/opensearch-gha-jobs/**
```

Comment:
I think this one already does exactly what you mentions, `test` and `deploy` are 2 different jobs.  The latter requires the former, and only run when it's approved and pushed to main.  Splitting the 2 jobs into 2 workflows is fine, but won't make this more secure I think (we can all run `make deploy` locally with our own key and bypass this altogether).

I basically copy/paste this workflow from https://github.com/pytorch/test-infra/blob/main/.github/workflows/log-classifier-lambda.yml which tests and deploys the log classifier.

Reply:
> Splitting the 2 jobs into 2 workflows is fine, but won't make this more secure I think (we can all run make deploy locally with our own key and bypass this altogether).

I would argue otherwise.  Even though we could deploy it locally, like you said, that's likely to be a deliberate decision by someone who has access to the AWS token (even we have to jump through some hoops to get that token). And people without that token cannot do a local deploy.

But with a workflow that automatically deploys on a PR being pushed, you get:
- Accidental deployments by sincere contributors (internal or external) who just wanted their PR reviewed, but didn't have enough confidence in the change to actually deploy it yet
- Potentially malicious deployments of unreviewed code by external contributors